src/macros.jl

```diff
@@ -309,18 +304,20 @@ end
 # end
 
 # three-argument constructconstraint! is used for two-sided constraints.
+constructconstraint!(v::Variable, lb::Real, ub::Real) = SingleVariableConstraint(v, MOI.Interval(lb, ub))
```

Comment:
What happens when `lb` and `ub` are integers, and the solver only supports `Interval{Float64}`?

Reply:
Currently this doesn't happen because they are initialized as floating point number
https://github.com/JuliaOpt/JuMP.jl/blob/e6c9f401b63afa630a0af720b6ed507bc0b97dc9/src/macros.jl#L434-L437
I'd say that it is the responsability of the caller of `constructconstraint!` to use relevant types for `lb` and `ub`.